xds/internal/client/client.go

```diff
@@ -304,14 +307,14 @@ type WeightedCluster struct {
 
 // HeaderMatcher represents header matchers.
 type HeaderMatcher struct {
-	Name         string      `json:"name"`
-	InvertMatch  *bool       `json:"invertMatch,omitempty"`
-	ExactMatch   *string     `json:"exactMatch,omitempty"`
-	RegexMatch   *string     `json:"regexMatch,omitempty"`
-	PrefixMatch  *string     `json:"prefixMatch,omitempty"`
-	SuffixMatch  *string     `json:"suffixMatch,omitempty"`
-	RangeMatch   *Int64Range `json:"rangeMatch,omitempty"`
-	PresentMatch *bool       `json:"presentMatch,omitempty"`
+	Name         string         `json:"name"`
+	InvertMatch  *bool          `json:"invertMatch,omitempty"`
+	ExactMatch   *string        `json:"exactMatch,omitempty"`
+	RegexMatch   *regexp.Regexp `json:"regexMatch,omitempty"`
```

Comment:
I don't know what JSON this field will be marshalled to.
But I don't think we actually care now.

We needed the json because the matchers were specified in balancer config, and passed to the routing balancer. Now that this is no longer marshaled to JSON, I think we can delete all the `json` tags.

We can do that in another PR.